tests/rest/admin/test_statistics.py

```diff
@@ -0,0 +1,463 @@
+            number_media: Number of media to be created for the user
+        """
+        upload_resource = self.media_repo.children[b"upload"]
+        for i in range(number_media):
+            # file size is 67 Byte
+            image_data = unhexlify(
+                b"89504e470d0a1a0a0000000d4948445200000001000000010806"
+                b"0000001f15c4890000000a49444154789c63000100000500010d"
+                b"0a2db40000000049454e44ae426082"
+            )
+
+            # Upload some media into the room
+            self.helper.upload_media(
+                upload_resource, image_data, tok=user_token, expect_code=200
+            )
+
+    def _check_fields(self, content: List[Dict[str, Any]]):
+        """Checks that all attributes are present in content
+        Args:
+            content: List that is checked for content
+        """
+        for c in content:
+            self.assertIn("user_id", c)
+            self.assertIn("displayname", c)
+            self.assertIn("media_count", c)
+            self.assertIn("media_length", c)
+
+    def _order_test(
+        self, order_type: str, expected_user_list: List[str], dir: Optional[str] = None
+    ):
+        """Request the list of users in a certain order. Assert that order is what
+        we expect
+        Args:
+            order_type: The type of ordering to give the server
+            expected_user_list: The list of user_ids in the order we expect to get
+                back from the server
+            dir: The direction of ordering to give the server
+        """
+
+        url = self.url + "?order_by=%s" % (order_type,)
+        if dir is not None:
+            if dir in ("b", "f"):
+                url += "&dir=%s" % (dir,)
+        request, channel = self.make_request(
+            "GET", url.encode("ascii"), access_token=self.admin_user_tok,
+        )
+        self.render(request)
+        self.assertEqual(200, channel.code, msg=channel.json_body)
+        self.assertEqual(channel.json_body["total"], len(expected_user_list))
+
+        returned_order = [row["user_id"] for row in channel.json_body["users_media"]]
+        self.assertListEqual(expected_user_list, returned_order)
+        self._check_fields(channel.json_body["users_media"])
+
+    def _set_displayname(self, user_id: str, accesss_token: str, displayname: str):
+        """Set a displayname of a specific user
+        Args:
+            user_id: user ID of the user whose displayname should be set
+            access_token: Access token of the user whose displayname should be set
+            displayname: new displayname to be set
+        """
+        request, channel = self.make_request(
+            "PUT",
+            "/profile/%s/displayname" % (user_id,),
+            content=json.dumps({"displayname": displayname}),
+            access_token=accesss_token,
+        )
+        self.render(request)
+        self.assertEqual(200, channel.code, msg=channel.json_body)
```

Comment:
Which `self.register_user` do you mean? Does it call https://github.com/matrix-org/synapse/blob/4fda58ddd257980bac8cfc277825f4deecc5dd7b/tests/unittest.py#L549-L557 ?
There is no chance to define a `displayname`.

Or do you mean I should implement this parameter at this point?

Reply:
It's done.

Reply:
Apologies, I misread what this function did and assumed it called https://github.com/matrix-org/synapse/blob/develop/docs/admin_api/user_admin_api.rst#create-or-modify-account, which did already have a `displayname` parameter. I probably wouldn't have suggested it otherwise as it was considerably more work for you.

However, thank you for adding it to the shared secret registration API as well - that will benefit others even more!